### PR TITLE
Adapter Serialisation: DataType now serialises properly

### LINKED ISSUE #146: Enable adapter workflows with varying "datatype" types
#### Description:

DataType objects can be dynamic with fields varying based on what "type" it represents (yay Python). For C# this gets tricky, but by having a GenericDataType and a DataType we should be able to handle both forms.

### PR DESCRIPTION
**Pull request by Tom-Kingstone**
### Issues addressed by this PR

Closes #146 

Serialised objects of `"DataType"` and `"GenericDataType"` are now differentiated by the serialiser, so will now no longer return the error: `Objects of type DataType are not yet supported for conversion to a LadybugTools object.`

Also forced the json to be an array if it was not already.

### CODE REVIEW

File: LadybugTools_Adapter/Convert/MetaData/DataType.cs [hunk 1]
```diff
@@ -73,13 +73,22 @@ public static BH.oM.LadybugTools.DataType ToDataType(Dictionary<string, object>
         {
             Dictionary<string, object> returnDict = new Dictionary<string, object>
             {
-                { "type", "GenericDataType" },
                 { "name", dataType.Name },
                 { "data_type", dataType.Data_Type }
             };
+            
+            string type;
```
Comment (FraserGreenroyd) [inline LadybugTools_Adapter/Convert/MetaData/DataType.cs:80]:
```suggestion
            string type = "DataType";
```

File: LadybugTools_Adapter/Convert/MetaData/DataType.cs [hunk 2]
```diff
@@ -73,13 +73,22 @@ public static BH.oM.LadybugTools.DataType ToDataType(Dictionary<string, object>
...
-            if (dataType.BaseUnit != "")
+            if (dataType.Data_Type == "GenericType")
+            {
+                type = "GenericDataType";
                 returnDict.Add("base_unit", dataType.BaseUnit);
+            }
+            else
+            {
+                type = "DataType";
+            }
```
Comment (FraserGreenroyd) [inline LadybugTools_Adapter/Convert/MetaData/DataType.cs:90]:
```suggestion
```

File: LadybugTools_Adapter/Convert/ToBHoM.cs [hunk 3]
```diff
@@ -38,6 +38,14 @@ public static partial class Convert
...
         {
             string json = File.ReadAllText(jsonFile.GetFullFileName());
+            if (!json.StartsWith("["))
+            {
+                json = "[" + json;
+            }
+            if (!json.EndsWith("]"))
+            {
+                json = json + "]";
+            }
```
Comment (FraserGreenroyd) [inline LadybugTools_Adapter/Convert/ToBHoM.cs:48]:
```suggestion
            if (!json.StartsWith("["))
                json = "[" + json;
                
            if (!json.EndsWith("]"))
                json = json + "]";
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by Tom-Kingstone** [comment]
@BHoMBot check required

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance

**Pull request merged**